### PR TITLE
feat(api): Support Parquet as a query response format.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This changes adds support for the header `Accept: application/vnd.apacher.parquet`. If this header is set the response for metric and stream queries will response with a Parquet file.

A metrics response has the columns `timestamp`, `labels` and `value`. A stream response has the columns `timestamp`, `labels` and `line`.

The Parquet file can be queried with e.g. DuckDB:
render1734096279010-min

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: docs/sources/reference/loki-http-api.md [hunk 1]
```diff
@@ -438,6 +438,35 @@ curl -u "User:$API_TOKEN" \
...
+│ line        │ VARCHAR                  │
+
+and for metrics:
+
+│ column_name │       column_type        │
+│   varchar   │         varchar          │
+|-------------|--------------------------|
+│ timestamp   │ TIMESTAMP WITH TIME ZONE │
+│ labels      │ MAP(VARCHAR, VARCHAR)    │
+│ line        │ DOUBLE                   │
```
Comment (chaudum) [inline docs/sources/reference/loki-http-api.md:468]:
```suggestion
│ value       │ DOUBLE                   │
```

**Review by chaudum — COMMENTED** [review]
LGTM

File: docs/sources/reference/loki-http-api.md [hunk 2]
```diff
@@ -438,6 +438,35 @@ curl -u "User:$API_TOKEN" \
   --data-urlencode 'query=sum(rate({job="varlogs"}[10m])) by (level)' | jq
 ```
 
+### Response Format
```
Comment (chaudum) [inline docs/sources/reference/loki-http-api.md:441]: I'm not sure if this is the best place for this documentation. It feels like that the response schemas belong to their respective query type sections.

Comment (jeschkies) [reply docs/sources/reference/loki-http-api.md:441]: I agree. This is in the section but maybe not the right spot. @JStickler what do you think?

Comment (JStickler) [reply docs/sources/reference/loki-http-api.md:441]: Ooooh, that's a tough one.  I think that chaudum has the right idea, that the response schemas belong with their respective query types.

If you think this is the correct section, I'd at least move your addition up above the examples, either to line 300 or line 350.

**Review by chaudum — APPROVED** [review]
LGTM Waiting for @JStickler for docs review.

**Review by JStickler — APPROVED** [review]
>LGTM Waiting for @JStickler for docs review.

[docs team]  docs updates LGTM

**Pull request merged**